xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/exporter/exporter.js

```diff
@@ -176,36 +180,90 @@ require([
+                    checkedPages.push(XWiki.Model.serialize(pageReference));
+                    // In case the page coudl have children
+                    if (pageReference.getName() == 'WebHome') {
+                        // The node could be checked but not loaded
+                        if (node.state.loaded == false) {
+                            // In that case, we need to add all children using a wildcard.
+                            var spaceReference = new XWiki.EntityReference('%', XWiki.EntityType.DOCUMENT, pageReference.parent);
+                            checkedPages.push(XWiki.Model.serialize(spaceReference));
+                        }
+                        // Special behaviour for the XAR export
+                        if (url.params.format == 'xar') {
+                            // Also add the WebPreferences
+                            var webPreferencesReference = new XWiki.EntityReference('WebPreferences', XWiki.EntityType.DOCUMENT,
+                                pageReference.parent);
+                            checkedPages.push(XWiki.Model.serialize(webPreferencesReference));
+                        }
+                    }
+                }
+            // the node is unchecked
+            } else {
+                // if the node is pagination then we should set the otherPages attribute
+                if (node.data.type == "pagination") {
+                    otherPages.attr('value', 'false');
+                } else {
+                    var pageReference = XWiki.Model.resolve(node.data.id, XWiki.EntityType.DOCUMENT);
+                    uncheckedPages.push(XWiki.Model.serialize(pageReference));
+                    // In case the page coudl have children
+                    if (pageReference.getName() == 'WebHome') {
+                        // The node could be checked but not loaded
+                        if (node.state.loaded == false) {
+                            // In that case, we need to add all children using a wildcard.
+                            var spaceReference = new XWiki.EntityReference('%', XWiki.EntityType.DOCUMENT,
+                                pageReference.parent);
+                            uncheckedPages.push(XWiki.Model.serialize(spaceReference));
+                        }
+                    }
+                }
             }
+          });
+
+          var aggregatePageNames = function (arrayOfNames) {
+              var finalValue = "";
+              for (var i = 0; i < arrayOfNames.length; i++) {
+                  finalValue += arrayOfNames[i];
+
+                  if (i < arrayOfNames.length - 1) {
+                      finalValue += "&";
+                  }
+              }
+              return finalValue;
+          };
+
+          checkedPagesInput.attr('value', aggregatePageNames(checkedPages));
+          uncheckedPagesInput.attr('value', aggregatePageNames(uncheckedPages));
```

Comment:
Again, better use ``val()``. What happens if a page name contains ``&``. Any character is valid within the page name.

Reply:
Yes, that should work.